Source/Synthesis.Solr/ContentSearch/SynthesisSolrDocumentTypeMapper.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Sitecore.ContentSearch.SolrProvider.Mapping;
+using Sitecore.ContentSearch.Linq.Common;
+using Sitecore.ContentSearch.Linq.Methods;
+using Sitecore.ContentSearch.Security;
+using Sitecore.Diagnostics;
+using Synthesis.ContentSearch;
+using System.Collections;
+using System.Text;
+
+namespace Synthesis.Solr.ContentSearch
+{
+	public class SynthesisSolrDocumentTypeMapper : SolrDocumentPropertyMapper
+	{
+		private readonly SynthesisDocumentTypeMapper _synthesisMapper = new SynthesisDocumentTypeMapper();
+		public override TElement MapToType<TElement>(Dictionary<string, object> document, SelectMethod selectMethod, IEnumerable<IFieldQueryTranslator> virtualFieldProcessors, IEnumerable<IExecutionContext> executionContexts, SearchSecurityOptions securityOptions)
+		{
+			ConvertDocumentToSythesisStandardTemplate(document);
+			var mappingResult = _synthesisMapper.MapToType<TElement>(() => ExtractFieldsFromDocument(document, virtualFieldProcessors), selectMethod);
+
+			// if the result type is not IStandardTemplateItem, use the default functionality
+			if (!mappingResult.MappedSuccessfully)
+				return base.MapToType<TElement>(document, selectMethod, virtualFieldProcessors, executionContexts, securityOptions);
+
+			return mappingResult.Document;
+		}
+		/// <summary>
+		/// there are some differences in how SOLR stores the standard fields, thus we need to convert a few of the solr fields
+		/// into something that synthesis will recognize
+		/// </summary>
+		/// <param name="document">the Solr results document</param>
+		private void ConvertDocumentToSythesisStandardTemplate(Dictionary<string, object> document)
+		{
+			string textSuffix = (Sitecore.Context.Site == null || Sitecore.Context.Language.Name == Sitecore.Context.Site.Language)? "_t" : "_t_"+Sitecore.Context.Language.Name.ToLower();
+			if (document.ContainsKey("__smallcreateddate_tdt"))
+				document["__smallcreateddate"] = document["__smallcreateddate_tdt"];
+			if (document.ContainsKey("__smallupdateddate_tdt"))
+				document["__smallupdateddate"] = document["__smallupdateddate_tdt"];
+			if (document.ContainsKey("__display_name" + textSuffix))
+				document["__display_name"] = document["__display_name" + textSuffix];
+			if (document.ContainsKey("__created_by_s"))
+				document["parsedcreatedby"] = document["__created_by_s"];
+			//updatedby is not available in the solr index
```

Comment:
I see what you mean, like `SolrStandardTemplateItem`. However, we couldn't do this even if we wanted to because `IndexFieldAttribute` is marked as inherited and allowMultiple = false, which means a subclass of `IStandardTemplateItem` cannot change the `IndexField` defined in it.

I wonder if there's a way we could inject ourselves between `IndexField` and SOLR, such that we could add and strip the suffix there and leave the generated model without the extensions? If not, are the rest of the field values in `IStandardTemplateItem` needing any mappings here? Like `parsedupdatedby` or `_group`?